compiler/src/main/java/org/qbicc/graph/InstanceOf.java

```diff
@@ -4,20 +4,23 @@
 
 import org.qbicc.type.BooleanType;
 import org.qbicc.type.ObjectType;
+import org.qbicc.type.ReferenceType;
 import org.qbicc.type.definition.element.ExecutableElement;
 
 /**
  * A node that represents a check of the upper bound of the value against the given type.
  */
-public final class InstanceOf extends AbstractValue implements InstanceOperation {
+public final class InstanceOf extends AbstractValue implements InstanceOperation, OrderedNode, BooleanValue {
```

Comment:
Ideally it would be nice if we could let InstanceOf nodes float more freely.  Is the OrderedNode addition needed because we eventually expand into a call in some cases or is it because of the BooleanValue and its interaction with phi nodes?

Should we open an issue to eventually get around to restoring these to not being OrderedNode?

Reply:
Right, converting it to control flow later causes some problems with scheduling.  We could fix the method call part using `callNoSideEffects` but the `if` stuff isn't as easy to fix (unless we use `select` instead, which is a possibility).